testsuite/tests/lib-runtime-events/test_corrupted.ml

```diff
@@ -0,0 +1,133 @@
+(* TEST
+ include runtime_events;
+ include unix;
+ set OCAML_RUNTIME_EVENTS_PRESERVE = "1";
+ libunix;
+ {
+   bytecode;
+ }{
+   native;
+ }
+*)
+
+  let runtime_begin _ _ _ = ()
+  let runtime_end _ _ _ = ()
+  let runtime_counter _ _ _ _ = ()
+  let alloc _ _ _ = ()
+  let lifecycle _ _ _ _ = ()
+  let lost_events _ _ = ()
+  let callbacks = Runtime_events.Callbacks.create
+    ~runtime_begin ~runtime_end ~runtime_counter ~alloc ~lifecycle ~lost_events ()
+
+  let parse path_pid =
+    let cursor =
+        Runtime_events.create_cursor path_pid in
+    let finally () = Runtime_events.free_cursor cursor in
+    Fun.protect ~finally @@ fun () ->
+    Runtime_events.read_poll cursor callbacks None
+
+  let parse_corrupted path_pid =
+    try let (_:int) = parse path_pid in ()
+    with Failure _ | Invalid_argument _ ->
+      (* parsing corrupted rings, raises exceptions,
+         this is expected *)
+      ()
+
+  let buf = Bytes.create (8 * 8)
+
+  let with_ring ring_file f =
+    let fd = Unix.openfile ring_file [Unix.O_RDWR] 0 in
```

Comment:
Could you add an explicit `O_CLOEXEC` or `O_KEEPEXEC`?

Reply:
I don't see an `exec` in this file, but just in case this code ends up being copy pasted elsewhere I'll add it.

Reply:
Right! I only spotted the `fork`… my bad.